packages/connector-node-v1/src/api.js

```diff
@@ -1,59 +1,28 @@
 import request from 'superagent';
 import JSZip from 'jszip';
-import { serializePromises } from './utils/common';
-import getMessage from '../translations';
+import { serializePromises, normalizeResource } from './utils/common';
 
 async function init(options) {
   options.onInitSuccess();
   options.onSignInSuccess();
 }
 
-function normalizeResource(resource) {
-  if (resource) {
-    return {
-      capabilities: resource.capabilities,
-      createdTime: Date.parse(resource.createdTime),
-      id: resource.id,
-      modifiedTime: Date.parse(resource.modifiedTime),
-      name: resource.name,
-      type: resource.type,
-      size: resource.size,
-      parentId: resource.parentId ? resource.parentId : null
-    };
-  } else {
-    return {};
-  }
-}
-
 async function getCapabilitiesForResource(options, resource) {
   return resource.capabilities || [];
 }
 
 async function getResourceById(options, id) {
-  let route = `${options.apiRoot}/files/${id}`;
-  let method = 'GET';
-  let response = await request(method, route).catch((error) => {
-    console.error(`Filemanager. getResourceById(${id})`, error);
-  });
-
-  let resource = response.body;
-
-  return normalizeResource(resource);
+  const route = `${options.apiRoot}/files/${id}`;
+  const method = 'GET';
+  const response = await request(method, route);
+  return normalizeResource(response.body);
 }
 
-async function getChildrenForId(options, { id, sortBy = 'name', sortDirection = 'ASC', onFail }) {
-  let route = `${options.apiRoot}/files/${id}/children?orderBy=${sortBy}&orderDirection=${sortDirection}`;
-  let method = 'GET';
-  let response = await request(method, route).catch((error) => {
-    console.error(`Filemanager. getChildrenForId(${id})`, error);
-    if (onFail) {
-      onFail({ message: getMessage(options.locale, 'unableReadDir') }) // TODO doesn't intercept for some reason
-    }
-  });
-
-  let rawResourceChildren = response.body.items;
-  let resourceChildren = await Promise.all(rawResourceChildren.map(async (o) => normalizeResource(o)));
-  return { resourceChildren };
+async function getChildrenForId(options, { id, sortBy = 'name', sortDirection = 'ASC' }) {
+  const route = `${options.apiRoot}/files/${id}/children?orderBy=${sortBy}&orderDirection=${sortDirection}`;
+  const method = 'GET';
+  const response = await request(method, route);
+  return { resourceChildren: response.body.items.map(normalizeResource) }
 }
 
 async function getParentsForId(options, id, result = []) {
```

Comment:
Replace **getParentsForId** name with **getParentsForResource** and arguments/calls accordingly